omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,244 @@
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("location");
+		description.addProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		String manualEntryEnabled = ((postBody.get("manualEntryEnabled") != null)
+		        && (!postBody.get("manualEntryEnabled").equals(""))) ? postBody.get("manualEntryEnabled").toString()
+		                : "false";
+		String automaticGenerationEnabled = ((postBody.get("automaticGenerationEnabled") != null)
+		        && (!postBody.get("automaticGenerationEnabled").equals("")))
+		                ? postBody.get("automaticGenerationEnabled").toString()
+		                : "false";
+		String sourceUuid = ((postBody.get("source") != null) && (!postBody.get("source").equals("")))
+		        ? postBody.get("source").toString()
+		        : null;
+		String identifierUuid = ((postBody.get("identifierType") != null) && (!postBody.get("identifierType").equals("")))
+		        ? postBody.get("identifierType").toString()
+		        : null;
+		String locationUuid = ((postBody.get("location") != null) && (!postBody.get("location").equals("")))
+		        ? postBody.get("location").toString()
+		        : Context.getLocationService().getDefaultLocation().getUuid();
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		IdentifierSource identifierSource = (identifierUuid != null)
+		        ? identifierSourceService.getIdentifierSourceByUuid(sourceUuid)
+		        : null;
+		if (identifierSource == null) {
+			throw new ResourceDoesNotSupportOperationException(
+			        "Creating an autoGenerationOption with missing parameters is not supported");
+		}
+		PatientIdentifierType patientIdentifierType = (identifierUuid != null)
+		        ? Context.getPatientService().getPatientIdentifierTypeByUuid(identifierUuid)
+		        : null;
+		if (patientIdentifierType == null) {
+			throw new ResourceDoesNotSupportOperationException(
+			        "Creating an autoGenerationOption with missing parameters is not supported");
+		}
+		Location location = Context.getLocationService().getLocationByUuid(locationUuid);
+		autoGenerationOption.setSource(identifierSource);
+		autoGenerationOption.setIdentifierType(patientIdentifierType);
+		autoGenerationOption.setManualEntryEnabled(Boolean.parseBoolean(manualEntryEnabled));
+		autoGenerationOption.setAutomaticGenerationEnabled(Boolean.parseBoolean(automaticGenerationEnabled));
+		autoGenerationOption.setLocation(location);
```

Comment:
Check if location is null before setting it

Reply:
done